third_party/move/move-vm/runtime/src/storage/environment.rs

```diff
@@ -162,15 +168,27 @@ impl RuntimeEnvironment {
         &self,
         compiled_module: Arc<CompiledModule>,
         module_size: usize,
+        module_hash: [u8; 32],
     ) -> VMResult<PartiallyVerifiedModule> {
-        // TODO(loader_v2): In loader V1, we also have a paranoid check, and a call to check_natives.
-        move_bytecode_verifier::verify_module_with_config(
-            &self.vm_config().verifier_config,
-            compiled_module.as_ref(),
-        )?;
-        if let Some(verifier) = &self.verifier_extension {
-            verifier.verify_module(compiled_module.as_ref())?;
+        if !VERIFIED_MODULES_V2.contains(&module_hash) {
+            // For regular execution, we cache already verified modules. Note
+            // that this even caches verification for the published modules.
+            // This should be ok because as long as the hash is the same, the
+            // deployed bytecode and any dependencies are the same, and so the
```

Comment:
it has the same dependencies, but are the dependencies the same?

Reply:
If dependencies are not the same, they are part of the published module bundle, and will be verified? We cache only local verification results, so it should be ok?

Reply:
yes, it's only the local check, it's all good, I didn't write properly, I meant maybe the comment is slightly ambiguous.

Reply:
@zekun000 asked the same question on big PR, let's also see what @runtian-zhou has to say